packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts

```diff
@@ -623,6 +623,12 @@ export class AnchoredRegion extends FoundationElement {
      *  Gets the anchor element by id
      */
     private getAnchor = (): HTMLElement | null => {
+        const rootNode = this.getRootNode();
+
+        if (rootNode instanceof ShadowRoot) {
+            return rootNode.getElementById(this.anchor);
+        }
+
         return document.getElementById(this.anchor);
```

Comment:
Would this be a problem if the anchored region is inside another component but the anchor isn't. Would writing it like this make it a little more robust?

```suggestion
        return this.getRootNode().getElementById(this.anchor) || document.getElementById(this.anchor);
```

Reply:
I don't think that makes a difference for nested shadow doms and I'm not sure how to do that and safely make typeScript happy.  I'd need to do something like `(this.getRootNode() as ShadowRoot)` which feels sketch without some type checking.